### DISCUSSION TITLE
Hi! I have a quesition with sionna ray tracing!!

### DISCUSSION THREAD

**Discussion by Howard-Rhee** [NVlabs/sionna#229] [Q&A] [answered]
I'm a student who is very interested in research using Sionna and I was wondering
 if it is possible to calculate things like RSRP and RSRQ using Sionna. 
I've looked at the documentation and it doesn't say anything about it, 
but I'm just wondering if I'm missing something. 
Can anyone tell me?

**Comment by jhoydis** [comment, marked as answer]
Hi,

Yes, you can definitely calculate RSRP and RSRQ based on either channel impulse responses or received signals. However, you would need to implement this yourself. Do you have a specific definition of RSRP and RSRQ in mind? If it is specific to a standard, you would probably need to implement also the required reference signals.

**Reply by Howard-Rhee** [reply]
Thank you for your kind response. I think your answer is enough to solve the problem.

**Reply by YuruZhang22** [reply]
I also want to get the RSRP and RSRQ from sionna. But I don't know how to calculate them based on CIR. Could you give me some tips?

**Reply by jhoydis** [reply]
RSRP simply relates to the total received power. To compute it, you could take the CIR, compute the discrete complex baseband-equivalent CIR from it (using [``cir_to_time_channel``](https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.cir_to_time_channel) and then simply sum the squared amplitudes of all channel taps.  You would then typically convert this number to dB and add the transmit power (in dB). However, the exact calculation depends on a standard.

RSRQ is somewhat equivalent to the SINR and requires the computation of the interference you might have. You would need to setup some link-level simulations to compute this value. It cannot be simply compute from the CIR alone.
